src/lapack/backends/cusolver/cusolver_task.hpp

```diff
@@ -43,6 +43,7 @@ static inline void host_task_internal(H &cgh, sycl::queue queue, F f) {
 template <typename H, typename F>
 static inline void onemkl_cusolver_host_task(H &cgh, sycl::queue queue, F f) {
     (void)host_task_internal(cgh, queue, f);
+    queue.wait(); //temporary workaround for issue #216
```

Comment:
Unfortunately this does not work because it is actually syncing the queue before the command group associated with this `onemkl_cusolver_host_task` is submitted: The result is that I find this does not solve #216

Reply:
Interestingly I find that replacing `queue.wait()` with a change to `host_task_internal`:

```
static inline void host_task_internal(H &cgh, sycl::queue queue, F f) {
    cgh.interop_task([f, queue](sycl::interop_handler ih) {
        auto sc = CusolverScopedContextHandler(queue, ih);
        f(sc);
        auto handle = sc.get_handle(queue);
        cusolverStatus_t err;
        cudaStream_t currentStreamId;
        CUSOLVER_ERROR_FUNC(cusolverDnGetStream, err, handle, &currentStreamId);
        cuStreamSynchronize(currentStreamId);
    });
}
```

is also not solving #216. It seems that you have to either call cuStreamSynchronize from within the user facing `onemkl_cusolver_host_task` or call `queue.wait()` after the command group is submitted.